File: 167_two_sum_sorted/TwoSum.cpp
class TwoSum {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0;
        int r = numbers.size() - 1;

        while (l < r) {
            if (numbers[l] + numbers[r] == target) {
                return {l + 1, r + 1};
            }
            if (numbers[l] + numbers[r] > target) {
                r--;
            }
             if (numbers[l] + numbers[r] < target) {
                l++;
            }
        }

       return {l + 1, r + 1};
    }
};

File: 167_two_sum_sorted/TwoSum.rs
impl TwoSum {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut l = 0;
        let mut r = numbers.len() - 1;

        while l < r {
            if numbers[l] + numbers[r] == target {
                return vec![(l + 1) as i32, (r + 1) as i32];
            }

            if numbers[l] + numbers[r] > target {
                 r = r - 1;
            }

            if numbers[l] + numbers[r] < target {
                 l = l + 1;
            }
        }

         vec![0, 0]
    }
}